ports/esp32s2/common-hal/os/__init__.c

```diff
@@ -57,5 +59,15 @@ mp_obj_t common_hal_os_uname(void) {
 }
 
 bool common_hal_os_urandom(uint8_t* buffer, uint32_t length) {
-    return false;
+    uint32_t i = 0;
+    while (i < length) {
+        uint32_t new_random = esp_random();
```

Comment:
It looks like this is only truly random when WiFi or Bluetooth is on: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/system.html#random-number-generation

Do you think it's worth accounting for that?

Reply:
I noticed that but I wasn't sure what the best philosophy here would be. If you know a flag I can check to see whether Wifi or Bluetooth is on, some `bool is_wireless_on()` equivalent, I could have this return false until it's true. Or I could throw a custom error to make it clear that they can get TRNG by enabling wireless.

Reply:
I'm not aware of a flag. I'm a bit surprised there isn't a way to just ask random about it. I think this is fine for now because we'll fall back to a non-random seed anyway.

Reply:
I've submitted a question to Espressif about it but I'll just leave it as is for now. I'm also surprised there isn't a clear way to tell if it's supplying "real" TRNG or not, seems pretty unsafe.